File: src/main/java/me/jameschan/burrow/kernel/command/entry/NewCommand.java
package me.jameschan.burrow.kernel.command.entry;

import java.util.Map;
import me.jameschan.burrow.kernel.command.Command;
import me.jameschan.burrow.kernel.common.ExitCode;
import me.jameschan.burrow.kernel.context.RequestContext;
import me.jameschan.burrow.kernel.furniture.annotation.CommandType;
import picocli.CommandLine;

@CommandLine.Command(name = "new", description = "Create a new entry and print the new try.")
@CommandType(CommandType.ENTRY)
public class NewCommand extends Command {
  public NewCommand(final RequestContext context) {
    super(context);
  }

  @Override
  public Integer call() {
    final var entry = context.getHoard().create(Map.of());
    buffer.append(context.getFormatter().format(entry));

    return ExitCode.SUCCESS;
  }
}

File: src/main/kotlin/carton/hay/command/NewCommand.kt
package burrow.carton.hay.command

import burrow.carton.hay.Hay
import burrow.carton.hay.MultipleAbsolutePathsException
import burrow.kernel.command.Command
import burrow.kernel.command.CommandData
import picocli.CommandLine
import picocli.CommandLine.ExitCode
import picocli.CommandLine.Parameters
import java.util.concurrent.atomic.AtomicInteger

@CommandLine.Command(
    name = "new",
    description = ["Creates a new path pair."]
)
class NewCommand(data: CommandData) : Command(data) {
    @Parameters(
        index = "0",
        paramLabel = "<relative-path>",
        description = ["The relative path to add."],
    )
    private var relativePath = ""

    @Parameters(
        index = "1",
        paramLabel = "<absolute-path>",
        description = ["The associated absolute path."],
        defaultValue = ""
    )
    private var absolutePath = ""

    override fun call(): Int {
        val hay = use(Hay::class)
        if (absolutePath.isNotBlank()) {
            hay.createEntry(relativePath, absolutePath)

            return dispatch(InfoCommand::class, listOf(relativePath))
        }

        try {
            hay.createEntry(relativePath)
        } catch (ex: MultipleAbsolutePathsException) {
            stderr.println("Multiple git repositories found: ")

            val index = AtomicInteger(0)
            ex.candidateAbsolutePaths.forEach {
                stderr.println("$[${index.getAndIncrement()}] $it")
            }

            return ExitCode.USAGE
        }

        return dispatch(InfoCommand::class, listOf(relativePath))
    }
}